### PR TITLE
Fixes #527: Added verify email address feature

### LINKED ISSUE #527: Allow users to verify email address
### Steps to reproduce
- go to navigation drawer -> select your profile picture

### Expected behaviour
- IF the email id is not verified, the cross is red. 
- If user selects red icon, open up a dialog/fragment : 
  title : `Your email id is not verified` 
  add a button : `Send verification email!`
- When user selects button, show a progress icon, and call the endpoint
- after the response is received, show the otp view (similar to that of reset password). Also a text at bottom : `Didn't receive OTP. Send mail again?`
- after user has entered OTP, call API to verify it

### Endpoint
`generate-verification-code`
`confirm-verification-code/<str:verification_code>`

### PR DESCRIPTION
**Pull request by Balram02**
## Description
Added email verification option.

Fixes #527 

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?
- [x] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: Android/app/src/main/java/io/github/project_travel_mate/ProfileActivity.java [hunk 1]
```diff
@@ -168,6 +174,26 @@ protected void onCreate(Bundle savedInstanceState) {
             updateOptionsMenu();
         }
 
+        isVerified.setOnClickListener(view -> {
+            AlertDialog.Builder builder = new AlertDialog.Builder(ProfileActivity.this);
+            if (!mIsVerified) {
+                builder.setTitle("Your email id is not verified");
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/ProfileActivity.java]: Please dont add constant strings here. These should be in strings.xml file

File: Android/app/src/main/java/io/github/project_travel_mate/ProfileActivity.java [hunk 2]
```diff
@@ -168,6 +174,26 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                builder.setPositiveButton("Verify Now", (dialogInterface, i) -> {
+                    sendVerificationEmail();
+                });
+                builder.setNegativeButton("Later", (dialogInterface, i) -> {
+                    dialogInterface.dismiss();
+                });
+            } else {
+                builder.setTitle("Your email id is verified");
+                builder.setPositiveButton("Ok", (dialogInterface, i) -> {
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/ProfileActivity.java]: move all the strings to strings.xml

**Comment by Balram02** [comment]
Here are the snaps..
<div>





></div>

**Comment by Swati4star** [comment]
Looks amazing!!! <3

**Pull request merged**